schema205/src/performance_map_base.h

```diff
@@ -0,0 +1,85 @@
+
+public:
+    performance_map_base() = default;
+    virtual ~performance_map_base() = default;
+    performance_map_base(const performance_map_base& other) = default;
+    performance_map_base& operator=(const performance_map_base& other) = default;
+
+  // ----------------------------------------------------------------------------------------------
+  /// @brief	
+  /// @param	j
+  // ----------------------------------------------------------------------------------------------
+    virtual void Initialize(const nlohmann::json& j) = 0;
+
+  // ----------------------------------------------------------------------------------------------
+  /// @brief	
+  /// @param	axis TBD
+  // ----------------------------------------------------------------------------------------------
+    inline void Add_grid_axis(std::vector<double>& axis) {
+        _grid_axes.emplace_back(Btwxt::GridAxis(axis));
+    }
+
+  // ----------------------------------------------------------------------------------------------
+  /// @brief	
+  /// @param	axis TBD
+  // ----------------------------------------------------------------------------------------------
+    inline void Add_grid_axis(std::vector<int>& axis) {
+        _grid_axes.emplace_back(Btwxt::GridAxis(std::vector<double>(axis.begin(), axis.end())));
+    }
+
+  // ----------------------------------------------------------------------------------------------
+  /// @brief	
+  /// @param	table TBD
+  // ----------------------------------------------------------------------------------------------
+    inline void Add_data_table(std::vector<double>& table) {
+        _btwxt.add_value_table(table);
+    }
+    
+  // ----------------------------------------------------------------------------------------------
+  /// @brief	
+  // ----------------------------------------------------------------------------------------------
+    inline void Finalize_grid() {
+        auto gd = Btwxt::GriddedData(_grid_axes);
+        _btwxt = Btwxt::RegularGridInterpolator(gd);
+    }
+
+  // ----------------------------------------------------------------------------------------------
+  /// @brief	
+  /// @param	table_index TBD
+  // ----------------------------------------------------------------------------------------------
+    inline double Calculate_performance(const std::vector<double> &target,
+                                        std::size_t table_index) {
+        return _btwxt.get_value_at_target(target, table_index);
+    }
+
+  // ----------------------------------------------------------------------------------------------
+  /// @brief	Using pre-populated grid axes and lookup tables, calculate a set of performance
+  ///         results.
+  /// @param	target 
+  // ----------------------------------------------------------------------------------------------
+    inline std::vector<double> Calculate_performance(const std::vector<double> &target) {
+        return _btwxt.get_values_at_target(target);
+    }
```

Comment:
@tanaya-bigladder can we overload this function for specific performance maps such that the arguments are the actual grid_variables and it returns a struct of the lookup_variables? For example:

`LookupStruct Calculate_performance(const double drybulb_temperature, const double relative_humidity)`

This would be used like: `cap_95 = Calculate_performance(95, 0.4).total_capacity`

The function would create the vector internally to call Btwxt.

Reply:
What if the return value stays a std::vector<double>? We already have enums in the LookupVariables classes to enumerate the lookup variables, and those would give named indices into the vector:

Reply:
That's what we are already doing. It works, but it's just a bit clunky and verbose. Are there downsides to offering both options?

Reply:
Not really. I can add new data structures for LookupVars, though are square brackets clunkier than adding a whole structure? It wasn't clear whether the overarching goal was the separate the inputs or the outputs (or both).

Reply:
Added.